tests/optim/utils/image/common.py

```diff
@@ -8,7 +8,182 @@
 from tests.optim.helpers import numpy_common
 
 
-class TestGetNeuronPos(unittest.TestCase):
+class TestMakeGridImage(BaseTest):
+    def test_make_grid_image_single_tensor(self) -> None:
+        test_input = torch.ones(6, 1, 2, 2)
+        test_output = common.make_grid_image(
+            test_input, nrow=3, padding=1, pad_value=0.0
+        )
+        expected_output = torch.tensor(
+            [
+                [
+                    [
+                        [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
+                        [0.0, 1.0, 1.0, 0.0, 1.0, 1.0, 0.0, 1.0, 1.0, 0.0],
+                        [0.0, 1.0, 1.0, 0.0, 1.0, 1.0, 0.0, 1.0, 1.0, 0.0],
+                        [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
+                        [0.0, 1.0, 1.0, 0.0, 1.0, 1.0, 0.0, 1.0, 1.0, 0.0],
+                        [0.0, 1.0, 1.0, 0.0, 1.0, 1.0, 0.0, 1.0, 1.0, 0.0],
+                        [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
+                    ]
+                ]
+            ]
+        )
+        self.assertEqual(list(expected_output.shape), [1, 1, 7, 10])

```

Comment:
The transformation doesn't seem very clear. Do you mind adding more details how 6 x 1 x 2 x 2 transforms into 1 x 1 x 7 x 10 ? `nrow` argument looks a bit unclear to me.

Reply:
The is 7 x 10 because have 3, 2 by 2 tensors in each row, which gives us two rows as we have 6 tensors in total. For this test we have 3 * 2 rows, which gives us 6 tensors with a border of size 1. For the vertical portion the border has a total size of 3, while there are 4 border values horizontally. The border doesn't double in size when it separates two tensors, and thus the values are not the same as if we simply padded the inputs with a border and then concatenated them.

Reply:
```
(3* 1) + 1+ (3* 2) = 10
(2* 1) + 1+ (2* 2) = 7

# Border has a size of 1, and we are using 3 in a row:
total_w = (n_row * border) + border + (n_row * tensor_dim_w)
total_h = (n_column * border) + border + (n_column * tensor_dim_h)
```

Reply:
I've added the above comment to the tests!

Reply:
I've added the above comment to the tests!